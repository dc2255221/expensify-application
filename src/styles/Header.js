import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { Colors, FontSize, Spacing } from '../constants';

export const StyledHeader = styled.header`
  background:  ${Colors.dark_blue};
`;

export const ContentContainer = styled.div`
  margin: 0 auto;
  padding: 0 ${Spacing.m_size};
  max-width: 80rem;
`;

export const HeaderContent = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: ${Spacing.s_size};
`;

export const StyledLink = styled(Link)`
  color: white;
  text-decoration: none;
`;

export const StyledH1 = styled.h1`
  margin: 0;
`;

export const StyledButton = styled.button`
  background: none;
  border: none;
  color: white;
  font-size: ${FontSize.large};
  font-weight: 300;
  padding: ${Spacing.s_size};
  cursor: pointer;
`;