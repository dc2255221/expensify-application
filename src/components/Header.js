import React from 'react';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';
import { StyledHeader, ContentContainer, HeaderContent, StyledLink, StyledH1, StyledButton } from '../styles/Header';

export const Header = ({ startLogout }) => (
    <StyledHeader>
      <ContentContainer>
        <HeaderContent>
          <StyledLink to="/dashboard"> 
            <StyledH1>Expensify</StyledH1> 
          </StyledLink>
          <StyledButton onClick={startLogout}> Logout </StyledButton>
        </HeaderContent>
      </ContentContainer>
    </StyledHeader>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);