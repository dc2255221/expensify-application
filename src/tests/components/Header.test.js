// react-test-renderer - simply utility for rendering; allows us to render our components inside regular JS code and we can assert what got rendered
// shallow rendering (only renders given component) vs full dom rendering (renders child components too)

import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json'; // able to get correct JSON snapshots to show up
import Header from '../../components/Header';

test('should render Header correctly', () => {
    const wrapper = shallow(<Header/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
    // expect(wrapper.find('h1').text()).toBe('Expensify');
});
