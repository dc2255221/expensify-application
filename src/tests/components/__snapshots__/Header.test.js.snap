// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<styled.header>
  <styled.div>
    <styled.div>
      <Styled(Link)
        to="/dashboard"
      >
        <styled.h1>
          Expensify
        </styled.h1>
      </Styled(Link)>
      <styled.button
        onClick={[Function]}
      >
         Logout 
      </styled.button>
    </styled.div>
  </styled.div>
</styled.header>
`;
